apps/web/src/components/Router.tsx

```diff
@@ -185,7 +185,14 @@ const createRoute = (locale: Locales, featureFlags: FeatureFlags) =>
             },
             {
               path: "logged-out",
-              loader: () => {
+              loader: ({ request }) => {
+                const url = new URL(request.url);
+                const from = url.searchParams.get("from");
+                if (from) {
+                  // eslint-disable-next-line @typescript-eslint/only-throw-error
+                  throw redirect(from);
```

Comment:
I think we need to do some validation here.  Otherwise we've got an [open redirect vulnerability](https://learn.snyk.io/lesson/open-redirect/?ecosystem=javascript).  I could sneakily send a user anywhere, like http://localhost:8000/en/logged-out?from=https%3A%2F%2Fbing.com

[In the authcallback](https://github.com/GCTC-NTGC/gc-digital-talent/blob/0fe07853292afa8298184d86cb2fbd970e989a0c/api/app/Http/Controllers/AuthController.php#L145) we only allow relative URLs but I'm not sure that will work here.

Reply:
But this will never work now, right?  Because it's set from the full URL?
` searchParams.append("from", window.location.href);`

Reply:
Even if we fix that, we want to be able to handle links in emails and things.  Instead of the very simple "does it start with a slash" logic, can we do something like check the host name or origin or something?

Reply:
okay, this is not complete (still need to update the pipeline YAML) but what do you think fo this?

[add test for full URL](https://github.com/GCTC-NTGC/gc-digital-talent/pull/14876/commits/a4a2926ac1f223ee9ece04a05865575c5ae8e5b1)

It should work like:

 - `/en` -> true (starts with `/`)
 - `talent.canada.ca` -> false (no protocol)
 - `http://github.com` -> false (does not match our hostname)
 - `https://talent.canada.ca` -> true (matches our hostname and has protocol)

Reply:
Hmm, I think this is on the right track but I don't think it will quite work.  If we use a build-time variable for this then the same artifact can't be used in dev, uat, and prod.  I don't love having to add new env variables, but if we wanted to do this it would have to be a deploy-time variable in index.html.  What if we just used a allow list that included dev and uat origins?   Or can we just check what the current location is and verify the it's either relative or to the same origin?

Reply:
That makes a lot of sense. Do you think this will work?

[check against current hostname instead of env var](https://github.com/GCTC-NTGC/gc-digital-talent/pull/14876/commits/c8dbc32083d464c02b800c72c86b024098369d84)